packages/taco/src/taco.ts

```diff
@@ -148,7 +148,7 @@ export const decrypt = async (
   porterUris: string[] = [],
   customParameters?: Record<string, CustomContextParam>,
 ): Promise<Uint8Array> => {
-  const porterUrisFull: string[] = getPorterUris(domain, porterUris);
+  const porterUrisFull: string[] = await getPorterUris(domain, porterUris);
```

Comment:
Wondering what the spirit of the API is if the caller already provides its list (1 or more) of porter URIs they want to use. 

In `getPorterUris` it seems we always append to the list, but perhaps we shouldn't if the user already provides a value. Obviously, since we are appending to a list, we still giving preference to their URIs first, but is it possible that the caller only wants to use specific Porters (for some reason 🤷 )?  This is not an option at the moment i.e. there is no ability for the user to actually override of whatever defaults we deem are good - only the ability to bias to their value(s) but not circumvent the use of our defaults if necessary. Don't get me wrong, in most cases I see devs not providing their own URIs but rather relying on our chain of URIs, but from an API perspective, it feels a little limiting.

So what if :
1. caller does not provide any porter Uris -> we get the chain of uris that includes defaults, and whatever is in the github repos
2. caller provides 1 or more porter Uris - we don't append and only use the URIs provided by the caller.

The `getPorterUris` call is available to the user as part of the `nucypher/shared` API so it can be used directly before calling `decrypt` if they desire. Not blindly appending may also save us from potential duplicate porter URIs if the user already provides existing default URIs directly.

Thoughts?

Reply:
from API point of view - I'd prefer append always to minimize amount of code for user. Other thought - it's possible to use `PorterClient` directly, in that case user can specify anything.

Reply:
I think the API behaviour I'd expect is that either the library uses the user-provided input, or it uses the defaults. While appending to the list of defaults is a possible option, I don't think it's very natural from a dev experience point of view.

Reply:
sure, changed api
✔️